x/evm/keeper/state_transition.go

```diff
@@ -355,13 +367,20 @@ func (k *Keeper) ApplyMessageWithConfig(
 	}
 
 	stateDB := statedb.NewWithParams(ctx, k, cfg.TxConfig, cfg.Params)
+	var evm *vm.EVM
 	if cfg.Overrides != nil {
 		if err := cfg.Overrides.Apply(stateDB); err != nil {
 			return nil, errorsmod.Wrap(err, "failed to apply state override")
 		}
 	}
+	if cfg.DebugTrace && cfg.BlockOverrides != nil {
+		blockCtx := k.NewEVMBlockContext(ctx, cfg)
+		cfg.BlockOverrides.Apply(&blockCtx)
+		evm = k.NewEVMWithBlockCtx(ctx, msg, cfg, stateDB, blockCtx)
+	} else {
+		evm = k.NewEVM(ctx, msg, cfg, stateDB)
+	}
```

Comment:
```suggestion
	blockCtx := k.NewEVMBlockContext(ctx, cfg)
	if cfg.BlockOverrides != nil {
		cfg.BlockOverrides.Apply(&blockCtx)
	}
	evm = k.NewEVMWithBlockCtx(ctx, msg, cfg, stateDB, blockCtx)
```

Reply:
we can even put the state/block overrides logic into the `NewEVM` function, since it has all the required information through the `cfg`.

Reply:
https://github.com/ethereum/go-ethereum/pull/26414
FYI, they have block overrides in eth_call as well

Reply:
also saw that, but need bump to 1.12.x

Reply:
Do we need their changes, we can simply add the parameter to eth_call api and pass all along to grpc handler